### PR TITLE
IP-346 - Migrate InterpretedGenomeRD

### PR DESCRIPTION
**Pull request by greglever**
[IP-346 - Migrate InterpretedGenomeRD](https://jira.extge.co.uk/browse/IP-346)
=========================
Migrate InterpretedGenomeRD from 3.0.0 to 4.2.0-SNAPSHOT as required by the CIP API to send Reported Variants to CVA 

Summary of Changes
==================
* Add `migrate_interpreted_genome_rd` to `MigrateReports3To420SNAPSHOT`
* Add `test_migrate_interpreted_genome_rd` to `TestMigrateReports3To420SNAPSHOT`
```
(.env) $ python -m unittest discover
..............................................
----------------------------------------------------------------------
Ran 46 tests in 2.550s

OK
(.env) $ 
```

### CODE REVIEW

File: protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py [hunk 1]
```diff
@@ -39,40 +47,61 @@ def migrate_interpretation_request_rd(self, old_interpretation_request_rd):
...
+        :rtype: reports_4_2_0_SNAPSHOT.InterpretedGenomeRD
+        """
+        new_interpreted_genome_rd = self.new_model.InterpretedGenomeRD.fromJsonDict(
+            jsonDict=old_interpreted_genome_rd.toJsonDict()
+        )
+
+        new_interpreted_genome_rd.interpretationRequestId = old_interpreted_genome_rd.InterpretationRequestID
+        new_interpreted_genome_rd.reportUrl = old_interpreted_genome_rd.reportURL
+        new_interpreted_genome_rd.reportUri = old_interpreted_genome_rd.reportURI
+
```
Comment (priesgo) [inline protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py]: Aren't we missing here migration from VersionControl to ReportVersionControl?

File: protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py [hunk 2]
```diff
@@ -39,40 +47,61 @@ def migrate_interpretation_request_rd(self, old_interpretation_request_rd):
...
+        new_interpreted_genome_rd.reportedStructuralVariants = self.migrate_reported_structural_variants(
+            old_reported_structural_variants=old_interpreted_genome_rd.reportedStructuralVariants
+        )
+
+        return self.validate_object(
+            object_to_validate=new_interpreted_genome_rd, object_type=self.new_model.InterpretedGenomeRD
+        )
 
     def migrate_report_event(self, old_report_event):
         new_report_event = self.new_model.ReportEvent.fromJsonDict(old_report_event.toJsonDict())
```
Comment (priesgo) [inline protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py]: Aren't we missing migration of GenomicFeature?

File: protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py (hunk 1 above)
Comment (greglever) [reply protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py]: Yup, good spot @priesgo 
I've made a commit here: https://github.com/genomicsengland/GelReportModels/commit/04523f7d928f5a3d7abe64abd8eab4576b7da2e6
What do you think of the explicit migration and the extra test ?

File: protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py (hunk 2 above)
Comment (greglever) [reply protocols/migration/migration_reports_3_0_0_to_reports_4_2_0_SNAPSHOT.py]: Good spot @priesgo !
I've made a commit here: https://github.com/genomicsengland/GelReportModels/commit/6f4fee158feae32c18e7b9a861d13cb360efd255
What do you think of the explicit feature and the extra test ?

**Comment by priesgo** [comment]
Gooooood. Thanks @greglever

**Pull request merged**